meilisearch-core/src/query_builder.rs

```diff
@@ -1,35 +1,45 @@
 use std::borrow::Cow;
+use std::collections::HashMap;
 use std::ops::{Range, Deref};
 use std::time::Duration;
 
+use either::Either;
+use sdset::SetOperation;
+
+use meilisearch_schema::FieldId;
+
 use crate::database::MainT;
-use crate::bucket_sort::{bucket_sort, bucket_sort_with_distinct};
-use crate::{criterion::Criteria, Document, DocumentId};
+use crate::bucket_sort::{bucket_sort, bucket_sort_with_distinct, SortResult};
+use crate::{criterion::Criteria, DocumentId};
 use crate::{reordered_attrs::ReorderedAttrs, store, MResult};
 use crate::facets::FacetFilter;
 
-use either::Either;
-use sdset::SetOperation;
-
-pub struct QueryBuilder<'c, 'f, 'd, 'fa, 'i> {
+pub struct QueryBuilder<'c, 'f, 'd, 'i> {
     criteria: Criteria<'c>,
     searchable_attrs: Option<ReorderedAttrs>,
     filter: Option<Box<dyn Fn(DocumentId) -> bool + 'f>>,
     distinct: Option<(Box<dyn Fn(DocumentId) -> Option<u64> + 'd>, usize)>,
     timeout: Option<Duration>,
     index: &'i store::Index,
-    facets: Option<&'fa FacetFilter>,
+    facet_fitlers: Option<FacetFilter>,
+    facets: Option<Vec<(FieldId, String)>>,
 }
 
-impl<'c, 'f, 'd, 'fa, 'i> QueryBuilder<'c, 'f, 'd, 'fa, 'i> {
+impl<'c, 'f, 'd, 'i> QueryBuilder<'c, 'f, 'd, 'i> {
```

Comment:
You removed the `'fa` lifetime here but not everywhere, is it normal or intended?